### ISSUE TITLE
Make `_convert_to_multiindex()` public and move to a general lib

### ISSUE THREAD

**Issue by DanilYachmenev** [kaizen-ai/kaizenflow#542]
The function is: https://github.com/sorrentum/sorrentum/blob/master/dataflow/system/source_nodes.py#L348

- [x] Make it public
- [x] Move to `dataflow/core/utils.py`
- [x] Add a simple unit test in `dataflow/core/test/test_utils.py`
    - Use this for inspiration: https://github.com/sorrentum/sorrentum/blob/master/helpers/test/test_hpandas.py#L1909
- [x] Update the calls in the codebase using the [script](https://github.com/sorrentum/sorrentum/blob/master/docs/Codebase_clean_up_scripts.md)

FYI @PomazkinG @samarth9008

**Comment by Dylan-Celli** [comment]
I can create the script replacement branch today, should I just link the corresponding script branch in a comment on this branch page or create a PR for this branch and link the script branch there? [codebase_clean_up_scripts.md](dataflow/core/utils.py)

**Comment by gpsaggese** [comment]
@DanilYachmenev this is another PR that should be integrated from `Sorrentum` to `cmamp` after it's done.
Also when we start doing these on-the-fly integration we might not need to use the replacement script.
I think the replacement script is needed only when the same function is used in many repos (e.g., we can check in `orange` as a canary)

**Comment by Dylan-Celli** [comment]
> I think the replacement script is needed only when the same function is used in many repos

If the script is unneeded then I can create a PR with my current changes

**Comment by DanilYachmenev** [comment]
@Dylan-Celli file PR whenever you have any changes to make
otherwise messages like the one you said are just getting lost and task is stuck for a week
at the end, it is your responsibility to deliver a solution

no need to use script, make it public by hand, update calls and add unit test

**Comment by Dylan-Celli** [comment]
> @Dylan-Celli file PR whenever you have any changes to make otherwise messages like the one you said are just getting lost and task is stuck for a week at the end, it is your responsibility to deliver a solution
> 
> no need to use script, make it public by hand, update calls and add unit test

Ok, I will do. I do still have some work to do before it is ready for a PR but I will just make it a PR with that question once I am done.

**Comment by rheenina** [comment]
@jainamp14 you can start doing this task from scratch if you’re blocked on my review

**Comment by jainamp14** [comment]
`class Test_convert_to_multiindex(hunitest.TestCase):

    def test_1(self):
        # Create a sample DataFrame
        data = {
            'end_time': ['2022-01-04 09:01:00-05:00', '2022-01-04 09:01:00-05:00', '2022-01-04 09:02:00-05:00'],
            'id': [13684, 17085, 13684],
            'close': [None, None, None],
            'volume': [0, 0, 0]
        }
        df = pd.DataFrame(data)

        # Call the function to convert the DataFrame
        result_df = dtfcorutil.convert_to_multiindex(df, 'id')

        # Assert that the resulting DataFrame has the expected structure
        expected_columns = pd.MultiIndex.from_tuples([(13684, 'close'), (17085, 'close'), (13684, 'volume'), (17085, 'volume')])
        self.assertTrue(result_df.columns.equals(expected_columns))`
        
This is my unit test and when running it with this command:
`pytest dataflow/core/test/test_utils.py::Test_convert_to_multiindex`
I am getting this error:
`FAILED dataflow/core/test/test_utils.py::Test_convert_to_multiindex::test_1 - NameError: name 'np' is not defined`
When I try to run the whole .py file everything is running and passing and I did import numpy as np so I am not sure what I am missing

**Comment by rheenina** [comment]
@jainamp14 seems like you are running not from docker container. run `invoke docker_bash` then try to re-run your test

**Comment by jainamp14** [comment]
I am running it from the docker container but just to make sure I got out and tried again and I got the same error.

**Comment by jainamp14** [comment]
I just found the bug using traceback I didn't import numpy in when I moved the function the test file is totally fine sorry

**Comment by Jd8111997** [comment]
@sonaalKant @samarth9008 While replacing the calls from the codebase, I found the redundant re implementation of the same function : ```_convert_to_multiindex()``` in /oms/notebooks/CmTask6678_crypto.com_universe_analysis.py. Shall I remove the function from this file ? 

Also, Do you want me to create a separate branch for updating and replacing the calls as mentioned in this [guide](https://github.com/sorrentum/sorrentum/blob/master/docs/work_tools/all.codebase_clean_up.how_to_guide.md) ?

**Comment by sonaalKant** [comment]
Lets raise a PR and make your changes accordingly. We will suggest you changes directly on the PR, it is much easier to review from PR with context.

**Comment by Jd8111997** [comment]
@sonaalKant @samarth9008 I sent a PR for this.

**Comment by samarth9008** [comment]
Can this be close now?

**Comment by Jd8111997** [comment]
@samarth9008 Yes, the PR has been merged.

**Comment by rheenina** [comment]
@samarth9008 check if changes are  in cmamp and can close if so

**Comment by Sameep2808** [comment]
> @Sameep2808 Can you check? High possibility it is already there as we did the integration two days ago

@samarth9008 Everything has been transferred to cmamp we can close the issue

**Comment by samarth9008** [comment]
Sg, thanks

Closing it.
